low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -214,6 +214,62 @@ def get_platform_sensors_info(self):
                 )
         return response
 
+    def get_nw_ports_info(self):
+        """Return the Network ports information."""
+        network_cable_data = []
+        io_counters = psutil.net_io_counters(pernic=True)
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            nic_info = self.get_health_template(interface, False)
+
+            specifics = {}
+            for addr in addrs:
+                if addr.family == socket.AF_INET:
+                    specifics["ipV4"] = addr.address
+
+            if interface in io_counters:
+                io_info = io_counters[interface]
+                specifics.update({
+                    "networkErrors": io_info.errin + io_info.errout,
+                    "droppedPacketsIn": io_info.dropin,
+                    "droppedPacketsOut": io_info.dropout,
+                    "packetsIn": io_info.packets_recv,
+                    "packetsOut": io_info.packets_sent,
+                    "trafficIn": io_info.bytes_recv,
+                    "trafficOut": io_info.bytes_sent
+                })
+
+            nw_status, nw_cable_conn_status = \
+                self.get_nw_status(interface)
+
+            specifics["nwStatus"] = nw_status
+            specifics["nwCableConnStatus"] = nw_cable_conn_status
+
+            # set health status, description and recommendation
+            map_status = {"UP": "OK", "DOWN": "Disabled/Failed",
+                          "UNKNOWN": "UNKNOWN"}
```

Comment:
Unknown should map to NA

Reply:
Cable status should be connected or disconnected.
And recommendation in this case, should not be default contact seagate support. If cable disconnection detected, just mention to try connecting back the cable and check again. same for nw port, to enable the port back & check again

Reply:
for the value `"nwCableConnStatus"`, we are referring to carrier status from `/sys/class/net/eth0/carrier` file, and possible values in that file are "UP" or "DOWN", should I consider it as "CONNETCTED" & "DISCONNECTED" ??
I am considering "NA "when we don't find the file on the node.
```
What: /sys/class/net/<iface>/carrier
Date: April 2005
KernelVersion: 2.6.12
Contact: netdev@vger.kernel.org
Description: Indicates the current physical link state of the interface.
 Posssible values are:
 == ===================== 
 0  physical link is down
 1  physical link is up
 == =====================
 Note: some special devices, e.g: bonding and team drivers will allow this attribute to be written to force a link state for operating correctly and designating another fallback interface.
```

Reply:
`   "description":"lo is in good health.", `
addressed.

Reply:
>possible values in that file are "UP" or "DOWN", should I consider it as "CONNETCTED" & "DISCONNECTED" ??

Yes

Reply:
Addressed the comment, updated the heath view in jira. 👍